[Outlook] (event-based activation) Reorganize documentation (#4324)
OfficeDev/office-js-docs-pr: docs/outlook/on-new-compose-events-walkthrough.md

---
title: Automatically set the subject of a new message or appointment
description: Learn how to implement an event-based add-in that automatically sets the subject of a new message or appointment.
ms.date: 12/19/2023
ms.topic: how-to
ms.localizationpriority: medium
---

# Automatically set the subject of a new message or appointment

> Need to insert a required disclaimer into all your messages and appointments? With an event-based add-in, the content is automatically added to new messages or appointments, so that you can focus on composing your message and still remain compliant with your organization's policies.

Reviewer: Suggestion to make the pitch statement a little shorter and more casual.
```suggestion
Need to add a required disclaimer to all your messages? With an event-based add-in, content is automatically added to new messages or appointments. Your users can focus on writing, instead of compliance.
```

> The following sections walk you through how to develop an add-in that handles the `OnNewMessageCompose` and `OnNewAppointmentOrganizer` events. By the end of this walkthrough, you'll have an add-in that automatically sets the subject of new messages and appointments being created.

Reviewer: Consider this, in case "walk" gets localized strangely.
```suggestion
The following sections teach you how to develop an add-in that handles the `OnNewMessageCompose` and `OnNewAppointmentOrganizer` events. By the end of this walkthrough, you'll have an add-in that automatically sets the subject of new messages and appointments being created.
```
